src/ThreadedExecutor.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright (C) 2018 European Spallation Source, ERIC. See LICENSE file */
+//===----------------------------------------------------------------------===//
+///
+/// \file
+///
+/// \brief Simple threaded executor to simplify the use of threading in this lib
+/// library.
+///
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <concurrentqueue/concurrentqueue.h>
+#include <functional>
+#include <future>
+#include <memory>
+#include <thread>
+
+class ThreadedExecutor {
+private:
+public:
+  using WorkMessage = std::function<void()>;
```

Comment:
Maybe `WorkMessage` -> `Job`, it is less overloaded and standard terminology?

Reply:
Does not sound (look) right to me when trying it. How about `JobType` and then rewrite `SendWork` into `SendWork(JobType Task)`?